js/usercss.js

```diff
@@ -195,11 +196,42 @@ var usercss = (() => {
         parseJSONValue(state);
         state.errorPrefix = '';
         if (Array.isArray(state.value)) {
-          result.options = state.value.map(text => createOption(text));
+          result.options = state.value.map(text => {
+            if (text.endsWith('*')) {
+              text = text.slice(0, -1);
+              defaultValue = text;
+            }
+            return createOption(text);
+          });
         } else {
-          result.options = Object.keys(state.value).map(k => createOption(k, state.value[k]));
+          result.options = Object.keys(state.value).map(k => {
+            if (k.endsWith('*')) {
+              state.value[k] = state.value[k].slice(0, -1);
+              k = k.slice(0, -1);
+              defaultValue = k;
+            }
+            return createOption(k, state.value[k]);
+          });
+        }
+        result.default = typeof defaultValue !== 'undefined' ? defaultValue : (result.options[0] || {}).name || '';
+        break;
+      }
+
+      case 'number':
+      case 'range': {
+        state.errorPrefix = 'Invalid JSON: ';
+        parseJSONValue(state);
+        state.errorPrefix = '';
+        // [default, start, end, step, units] (start, end, step & units are optional)
+        if (Array.isArray(state.value) && state.value.length) {
+          result.default = parseFloat(state.value.shift());
+          const nonDigit = /[^\d.+-]/;
+          // label may be placed anywhere after default value
+          const labelIndex = state.value.findIndex(item => nonDigit.test(item));
+          // but should not contain any numbers '4px' => 'px'
+          result.units = labelIndex < 0 ? '' : state.value.splice(labelIndex, 1)[0].toString().replace(/[\d.+-]/g, '');
+          result.range = state.value.filter(item => !nonDigit.test(item));
```

Comment:
You mean have the `@var` set as an object instead of an array? The array is a lot less typing.

```
@var range    imgHeight  "Max image height" [50, 10, 200, 10, "px"]
```

versus

```
@var range    imgHeight  "Max image height" {
  "default": 50,
  "min": 10,
  "max": 200,
  "step": 10,
  "units": "px"
}
```

Reply:
Nice... I cleaned it all up using this method.